modules/PheWAS.py

```diff
@@ -127,58 +157,76 @@ def main():
     with open(Config.PHEWAS_CATALOG_EVIDENCE_FILENAME, 'w') as outfile:
 
         with open(Config.PHEWAS_CATALOG_FILENAME) as r:
-            catalog = tqdm(csv.DictReader(r), total=TOTAL_NUM_PHEWAS)
-            for i, row in enumerate(catalog):
+            #catalog = tqdm(csv.DictReader(r), total=TOTAL_NUM_PHEWAS)
+            catalog = pd.read_csv(r, dtype={'basepair': str, 'phewas_code': str}).fillna(np.nan)
+            
+            # Parsing genes
+            catalog["gene"] = catalog["gene"].dropna().apply(lambda X: ensgid_from_gene(X, ensgid))
+            catalog.dropna(subset=["gene"], inplace=True)
+            # Merging dataframes: more records due to 1:many associations
+            enriched_catalog = pd.merge(catalog, phewas_w_consequences, on=["gene", "snp"])
+
+            one2many_variants = {i for i in phewas_w_consequences["snp"] if phewas_w_consequences["snp"].tolist().count(i) > 1}
+            enriched_catalog["variant_id"] = enriched_catalog.dropna(how="any", subset=["chrom", "pos", "ref", "alt"]).apply(lambda X: write_variant_id(X, one2many_variants), axis=1)
+            enriched_catalog.drop(["csq_arr", "most_severe_gene_csq"], axis=1, inplace=True)
+
+            # Dropping exploded duplicates
+            columns = ['chromosome', 'basepair', 'gene','snp', 'phewas_code', 'phewas_string', 'cases', 'odds_ratio','p','chrom','pos','most_severe_csq','consequence_link']
```

Comment:
This does not consider rsids with different positions or functional consequences as duplicates. However, given than the `one2many_variants` dictionary is only built for those ids, if that ever happened, their `variant_id` would be `None` so they would all share the same `variant_id` in the UAFs. If that happened, there would still be duplicates.

This situation is theoretically possible but I don't know if it happens with this data, so the question is whether it's worth putting the necessary safeguards in place to avoid the issues.

Reply:
After discussions, we decided no t to code for this situation as it isn't expected to happen in the PheWAS data and if it does the duplicates will arise when running the validator or the pipeline